CMakeLists.txt

```diff
@@ -2,93 +2,173 @@ cmake_minimum_required (VERSION 3.5)
 
-#  Configure HDF5
-find_package (HDF5 REQUIRED COMPONENTS Fortran)
-include_directories (${HDF5_INCLUDE_DIRS})
+#-------------------------------------------------------------------------------
+#  Sanitizer options
+#-------------------------------------------------------------------------------
+add_library (sanitizer INTERFACE)
+
+macro (register_sanitizer_option name)
+    string (TOUPPER ${name} upper_name)
+    
+    option (SANITIZE_${upper_name} "Enable the ${name} sanitizer" OFF)
+
+    target_compile_options (sanitizer
+                            INTERFACE
+                            $<$<BOOL:${SANITIZE_${upper_name}}>:-fsanitize=${name}>
+    )
+    target_link_options (sanitizer
+                         INTERFACE
+                         $<$<BOOL:${SANITIZE_${upper_name}}>:-fsanitize=${name}>
+    )
+endmacro ()
+
+register_sanitizer_option (address)
+register_sanitizer_option (leak)
+register_sanitizer_option (memory)
+register_sanitizer_option (thread)
+register_sanitizer_option (undefined)
+register_sanitizer_option (float-divide-by-zero)
+
+#-------------------------------------------------------------------------------
+#  Build Locations
+#-------------------------------------------------------------------------------
+#  FIXME: Leave this as is for now but check if we want to use the cmake install
+#         functions better.
 
-#  Configure EZspline
-option (USE_PSPLINE "Use PSPLINE interpolation" OFF)
-if (${USE_PSPLINE})
-   #include_directories (${EZSpline_INCLUDE_PATH})
-   #set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${EZSpline_LINK_FLAGS}")
-   add_definitions (-DPSPLINE)
-endif ()
+#  Set directories to build all binary files.
+set (EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR}/build/bin)
+set (CMAKE_Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/build/modules)
+set (CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/build/lib)
 
-#  Configure FIO interpolation
-option (USE_FIO "Use FIO interpolation support" OFF)
-if (${USE_FIO})
-    include_directories (${FIO_INCLUDE_PATH} ${NIMROD_INCLUDE_PATH} ${NIMFIO_INCLUDE_PATH} ${NIMCORE_INCLUDE_PATH} ${NIMLIB_INCLUDE_PATH} ${NIMEXT_INCLUDE_PATH})
-    set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${FIO_LINK_FLAGS} ${NIMFIO_LINK_FLAGS} ${NIMCORE_LINK_FLAGS} ${NIMLIB_LINK_FLAGS} ${NIMEXT_LINK_FLAGS}")
-    add_definitions (-DFIO)
-endif ()
+#-------------------------------------------------------------------------------
+#  Dependencies
+#-------------------------------------------------------------------------------
+find_package (OpenMP)
+find_package (MPI REQUIRED)
+find_package (HDF5 REQUIRED COMPONENTS Fortran)
 
+add_library (korc_depends INTERFACE)
+
+target_link_libraries (korc_depends
+
+                       INTERFACE
+
+                       sanitizer
+                       $<$<BOOL:${OpenMP_Fortran_FOUND}>:OpenMP::OpenMP_Fortran>
+                       $<$<BOOL:${OpenMP_CXX_FOUND}>:OpenMP::OpenMP_CXX>
```

Comment:
What's using this?

Reply:
https://github.com/ORNL-Fusion/KORC/blob/b3e9f3b81f6920d110f8f3510d18e9d4d5cd55ec/src/CMakeLists.txt#L36

Reply:
What is using `OpenMP::OpenMP_CXX`?